google/cloud/ndb/model.py

```diff
@@ -4743,6 +4870,87 @@ def __init__(_self, **kwargs):
         if projection:
             self._set_projection(projection)
 
+    def _get_property_for(self, p, indexed=True, depth=0):
+        """Internal helper to get the Property for a protobuf-level property."""
+        if isinstance(p.name(), six.text_type):  # pragma: NO PY2 COVER
+            p.set_name(bytes(p.name(), encoding="utf-8"))
+        parts = p.name().decode().split(".")
+        if len(parts) <= depth:
+            # Apparently there's an unstructured value here.
+            # Assume it is a None written for a missing value.
+            # (It could also be that a schema change turned an unstructured
+            # value into a structured one.  In that case, too, it seems
+            # better to return None than to return an unstructured value,
+            # since the latter doesn't match the current schema.)
+            return None
+        next = parts[depth]
+        prop = self._properties.get(next)
+        if prop is None:
+            prop = self._fake_property(p, next, indexed)
+        return prop
+
+    def _fake_property(self, p, next, indexed=True):
+        """Internal helper to create a fake Property. Ported from legacy datastore"""
+        # A custom 'meaning' for compressed properties.
+        _MEANING_URI_COMPRESSED = "ZLIB"
+        if hasattr(self, "_clone_properties"):
```

Comment:
Is this ever true?

Reply:
Not in any of the testing I did. The ported code in fact just assumed the attribute would be there, so I added this to avoid crashing. Unsure if a migrated legacy model could have it.

Reply:
The preceding underscore would imply that users weren't intended to override it. I could be convinced of a better safe than sorry stance, though.